stream.go

```diff
@@ -14,10 +14,15 @@ const (
 	streamSYNSent
 	streamSYNReceived
 	streamEstablished
-	streamLocalClose
-	streamRemoteClose
-	streamClosed
-	streamReset
+	streamFinished
+)
+
+type halfStreamState int
+
+const (
+	halfOpen halfStreamState = iota
+	halfClosed
```

Comment:
comment on the direction of which half?

Reply:
Either half. There's one state for each half.